packages/chat-client/src/controllers/engine.ts

```diff
@@ -856,9 +864,42 @@ export class ChatEngine extends IChatEngine {
         timestamp: new Date(decodedPayload.iat).getTime(),
       };
 
+      const identityKeyPub = await this.client.identityKeys.getIdentity({
+        account: selfAccount,
+      });
+
+      const iat = Date.now();
+      const receiptKeyClaims = {
+        iat,
+        exp: jwtExp(iat),
+        iss: encodeEd25519Key(identityKeyPub),
+        sub: hashMessage(message.message),
+        ksu: this.keyserverUrl,
+        aud: composeDidPkh(message.authorAccount),
+        act: "chat_receipt",
+      };
+
       this.setMessage(topic, message);
-      await this.sendResult<"wc_chatMessage">(payload.id, topic, true);
       this.client.emit("chat_message", { id, topic, params: message });
+
+      // If the author is a registered account,
+      // and the peer is not, then there is no reason to respond to the message
+      if (authorAccount === selfAccount) {
+        try {
+          await this.client.identityKeys.getIdentity({
+            account: receipientAccount,
+          });
+        } catch (e) {
```

Comment:
Should we just add a `this.client.logger.info('Peer is not registered, skipping message')` ?